### PR TITLE
Fix upstream pdf ocr hanging on blank page

### PR DESCRIPTION
**Pull request by darkdh**
When `PdfAccessibilityTreeBuilder::AddRemainingAnnotations` processes ocr requests, blank page will have size 0 images_ per page, which makes the ocr request unprocessed.
In `PdfOcrHelper::ReceiveOcrResultsForImage`, it requires `remaining_page_count_` equals zero in order for
`PdfAccessibilityTree::OnOcrDataReceived` to be called and then also for `PdfOcrHelper::AreAllPagesOcred()` to be satisfied in order to set IDS_PDF_OCR_COMPLETED or IDS_PDF_OCR_NO_RESULT otherwise it would be stuck in IDS_PDF_OCR_IN_PROGRESS forever.

Page images info is passed into ax tree builder from PdfViewWebPlugin::PrepareAndSetAccessibilityPageInfo()-> chrome_pdf::GetAccessibilityInfo()->PDFiumEngine::GetImageInfo()-> PDFiumPage::GetImageInfo()->PDFiumPage::CalculateImages() and passed `AccessibilityPageObjects` into
PdfAccessibilityTree::SetAccessibilityPageInfo()-> PdfAccessibilityTree::AddPageContent()

So we create an empty image for blank page in
PDFiumPage::CalculateImages() to fix the issue.

Resolves https://github.com/brave/brave-browser/issues/41113

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Load any pdf that contains a blank page.
2. Open Leo and do summarize.
3. There should be response instead of waiting forever.

### CODE REVIEW

File: chromium_src/pdf/pdfium/pdfium_page.cc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/pdfium/public/fpdf_catalog.h"
+#define FPDFCatalog_IsTagged    \
```
Comment (mkarolin) [inline chromium_src/pdf/pdfium/pdfium_page.cc:7]: Nit: empty line above

File: chromium_src/pdf/pdfium/pdfium_page.cc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  false;                        \
+  if (!page_object_count) {     \
+    images_.push_back(Image()); \
+    return;                     \
+  }                             \
+  is_tagged = FPDFCatalog_IsTagged
```
Comment (mkarolin) [inline chromium_src/pdf/pdfium/pdfium_page.cc:13]: Nit: empty line below

**Review by mkarolin — APPROVED** [review]
`chromium_src`, `DEPS` ++ (with nits).

File: chromium_src/pdf/pdfium/pdfium_page.cc (hunk 1 above)
Comment (darkdh) [reply chromium_src/pdf/pdfium/pdfium_page.cc:8]: fixed and squashed, thanks.

File: chromium_src/pdf/pdfium/pdfium_page.cc (hunk 2 above)
Comment (darkdh) [reply chromium_src/pdf/pdfium/pdfium_page.cc:14]: https://github.com/brave/brave-core/pull/25735#discussion_r1783323105

**Review by bbondy — APPROVED** [review]

**Pull request merged**